app/server/pkg/community/google/analytics.go

```diff
@@ -0,0 +1,137 @@
+package google
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/analytics/v3"
+)
+
+const (
+	timeInterval = 30 * time.Minute
+	viewID       = "ga:208521052"
+	startDate    = "2019-12-01"
+	endDate      = "today"
+	filters      = "ga:eventCategory!=key1"
+	base         = 10
+	bitSize      = 64
+)
+type CommGAResponse struct{
+	TotalRuns string `json:"total-runs"`
+	OpInstalls string `json:"operator-installs"`
+	CityData [][]string `json:"geo-city"`
+	CountryData [][]string `json:"geo-country"`
+	DailyData [][]string `json:"daily-data"`
+}
+// GAResponseJSONObject is the array of GAResponse struct
+var GAResponseJSONObject CommGAResponse
+
+// Handler is responsible for the looping the UpdateAnalyticsData()
+func Handler() {
+	for true {
+		log.Infof("Updating Google Analytics Data ...")
+		err := getGraphData()
+		if err != nil {
+			log.Error(err)
+		}
+		err = getTotalCounts()
+		if err != nil {
+			log.Error(err)
+		}
+		log.Infof("Updated Google Analytics Data ...")
+		time.Sleep(timeInterval)
+	}
+}
+
+func getterJWTConfig() (*http.Client, error) {
+	key, err := ioutil.ReadFile("/etc/analytics/auth.json")
+	if err != nil {
+		return nil, fmt.Errorf("Error while getting the auth.json file, err: %s", err)
+	}
+	jwtConf, err := google.JWTConfigFromJSON(key, analytics.AnalyticsReadonlyScope)
+	if err != nil {
+		return nil, fmt.Errorf("Error while setting the JWTConfig, err: %s", err)
+	}
+	httpClient := jwtConf.Client(oauth2.NoContext)
+	return httpClient, nil
+}
+
+func getterSVC(httpClient *http.Client) (*analytics.Service, error) {
+	svc, err := analytics.New(httpClient)
+	if err != nil {
+		return nil, fmt.Errorf("Error while setting up NewClient, err: %s", err)
+	}
+	return svc, nil
+}
+
+// UpdateAnalyticsData sends the GET request to the GA instance and receives the events' metrics at every t time intervals
+// and updates the global JSON object for containing the response
+func getTotalCounts() error {
```

Comment:
Total count method is also there in analytics package, can't we use that

Reply:
This method is different from the one in the analytics and also because of the reason stated above, the duplication is intentional.